pkg/cluster/delete.go

```diff
@@ -377,6 +378,77 @@ func (m *manager) deleteClusterMsiCertificate(ctx context.Context) error {
 	return err
 }
 
+func (m *manager) deleteIdentityFederations(ctx context.Context) error {
+	if m.doc.OpenShiftCluster.Properties.PlatformWorkloadIdentityProfile == nil {
+		return nil
+	}
+
+	if m.clusterMsiFederatedIdentityCredentials == nil {
+		m.log.Warning("cluster MSI federated identity credentials client is nil, trying to initialize")
+		err := m.initializeClusterMsiClients(ctx)
+		if err != nil {
+			m.log.Errorf("cluster MSI federated identity credentials client initialization failed with error: %v", err)
+			return nil
+		}
+	}
+
+	platformWIRolesByRoleName := m.platformWorkloadIdentityRolesByVersion.GetPlatformWorkloadIdentityRolesByRoleName()
+	platformWorkloadIdentities := m.doc.OpenShiftCluster.Properties.PlatformWorkloadIdentityProfile.PlatformWorkloadIdentities
+
+	for _, identity := range platformWorkloadIdentities {
+		_, exists := platformWIRolesByRoleName[identity.OperatorName]
+		if !exists {
+			continue
+		}
+
+		identityResourceId, err := azure.ParseResourceID(identity.ResourceID)
+		if err != nil {
+			return err
+		}
+
+		platformWIRole, exists := platformWIRolesByRoleName[identity.OperatorName]
+		if !exists {
+			continue
+		}
+
+		for _, sa := range platformWIRole.ServiceAccounts {
+			federatedIdentityCredentialResourceName, err := m.getPlatformWorkloadIdentityFederatedCredName(sa, identity)
+			if err != nil {
+				m.log.Errorf("failed to get federated identity credential name for %s: %v", identity.ResourceID, err)
```

Comment:
This is an error on the RP side and not related to the cluster MSI having been deleted by the cx, so I think we should error out here rather than logging the error and continuing.

Reply:
updated